### ISSUE TITLE
Flatpak support

### ISSUE THREAD

**Issue by rugk** [AtomLinter/linter-shellcheck#115]
Currently when Atom is used as a flatpak it cannot start other applications, as it is sandboxed. 

Here is one way how one could start shellcheck outside of the flatpak sandbox to analyse the file: https://unix.stackexchange.com/questions/412869/how-to-allow-gui-application-in-flatpak-to-run-cli-tool/422438#422438

Maybe this needs proper support in Atom generally?

**Comment by Arcanemagus** [comment]
Atom doesn't officially support flatpacks, so you would need to bring this up with wherever you got that from.

**Comment by rugk** [comment]
Actually, I found a way: [You can add the permissions like this](https://discussion.fedoraproject.org/t/goals-for-fedora-silverblue-30/773/25?u=rugk):

``sh
sudo flatpak override io.atom.Atom --talk-name=org.freedesktop.Flatpak
```

Then, inside of the flatpak you should be able to run it via `flatpak-spawn --host shellcheck`.
That is the idea, at least…

However, it does not work…

It still ends in something like this:
![flatpakissue](https://user-images.githubusercontent.com/11966684/54626516-97336b00-4a71-11e9-82bc-adfdeed789fe.png)

Maybe it tries to run the command as the full binary path? As it contains spaces, that is obviously hard… :wink:

**Comment by Arcanemagus** [comment]
You would need to modify this in order to call it in that manner, since the first argument is _only_ the binary file to run, the second argument is where the arguments to the executable go:
https://github.com/AtomLinter/linter-shellcheck/blob/99cbb3583509dc17f525331d310f564994fda071/lib/main.js#L79

**Comment by rugk** [comment]
Well… can't you somehow change it, so one can always configure it properly without code modifications?

**Comment by Arcanemagus** [comment]
There have been pretty hacky ways of doing that put into a few providers, but as I stated earlier Atom itself is **not** supported in a Flatpak environment. Since I personally have no way of replicating it I'm not interested in supporting it either.

**Comment by rugk** [comment]
AH, wait, I found a convenient workaround. You just create a "wrapper shell script", put it in some location that is accessible to the flatpak and use it as an option.

Shell script:
```sh
#!/bin/sh
flatpak-spawn --host shellcheck "$@"
```
